# Examine census response rates, with data at state and census tract levels
## Data from: https://2020census.gov/en/response-rates.html

In [112]:
import pandas as pd

# in excel changed original dates from m/d/yyyy to yyyy-mm-dd
initial_df = pd.read_csv('decennialrr2020_working.csv')
# had to resave as UTF-8 CSV, hence the 2
crosswalk = pd.read_csv('decennialrr2020_crosswalkfile2.csv')
# states paired with region as defined by census map at https://www2.census.gov/geo/pdfs/maps-data/maps/reference/us_regdiv.pdf
regions = pd.read_csv('state_region.csv')

## Columns


GEO_ID = Geographic Identifier

RESP_DATE = Posting Date

State = name of state (one of the 50 states, District of Columbia, Puerto Rico, or NaN)

Geo_Name = name of the tract, county, state

Region = region of the U.S. in which state is located as defined by census map at https://www2.census.gov/geo/pdfs/maps-data/maps/reference/us_regdiv.pdf

Geo_Type = type of geography; possible answers include Census Tract, Congressional District, Consolidated City, Country, County, County Subdivision, Place, Region, State, Tribal Tract, Tribal Area

DRRINT = Daily Self-Response Rate - Internet

DRRALL = Daily Self-Response Rate – Overall

CRRINT = Cumulative Self-Response Rate - Internet

not_int = new calculated column showing response rate NOT from internet

CRRALL = Cumulative Self-Response Rate – Overall

DINTMIN = Minimum Daily Internet Self-Response Rate

DMIN = Minimum Daily Overall Self-Response Rate

CINTMIN = Minimum Cumulative Internet Self-Response Rate

CMIN = Minimum Cumulative Overall Self-Response Rate

DINTMAX = Maximum Daily Internet Self-Response Rate

DMAX = Maximum Daily Overall Self-Response Rate

CINTMAX = Maximum Cumulative Internet Self-Response Rate

CMAX = Maximum Cumulative Overall Self-Response Rate

DINTAVG = Average Daily Internet Self-Response Rate

DAVG = Average Daily Overall Self-Response Rate

CINTAVG = Average Cumulative Internet Self-Response Rate

CAVG = Average Cumulative Overall Self-Response Rate

DINTMED = Median Daily Internet Self-Response Rate

DMED = Median Daily Overall Self-Response Rate

CINTMED = Median Cumulative Internet Self-Response Rate

CMED = Median Cumulative Overall Self-Response Rate

In [158]:
# merge responses and crosswalk
merged1 = pd.merge(initial_df, crosswalk, on='GEO_ID')

#merge merged1 with region data
merged = pd.merge(merged1, regions, on='State')

# create column showing responses not from internet
merged['not_int'] = merged.CRRALL - merged.CRRINT

#reorder columns to move State, Geo_Name and Geo_Type to front; also going to drop some values
cols = merged.columns.tolist()
cols = ['GEO_ID', 'RESP_DATE', 'State', 'Geo_Name', 'Region', 'Geo_Type', 
        'DRRINT', 'DRRALL', 'CRRINT', 'not_int', 'CRRALL']
merged = merged[cols]
merged

# save merged df to csv
#merged.to_csv('merged2020rates.csv')

## Census Tracts

In [136]:
# just census tract geo types
tracts = merged[merged['Geo_Type'].str.contains("Tract")]

# which tract has the highest response self-response rate on 6/9
tracts.sort_values(by='DRRALL', ascending=False)

GEO_ID RESP_DATE         State                Geo_Name  \
100700  1400000US46037952800  6/9/2020  South Dakota         Tract 9528, Day   
26189   1400000US13121003700  6/9/2020       Georgia        Tract 37, Fulton   
24103   1400000US12119910800  6/9/2020       Florida      Tract 9108, Sumter   
42486   1400000US20125950800  6/9/2020        Kansas  Tract 9508, Montgomery   
100720  1400000US46059975700  6/9/2020  South Dakota        Tract 9757, Hand   
...                      ...       ...           ...                     ...   
49248   1400000US24047951300  6/9/2020      Maryland   Tract 9513, Worcester   
49250   1400000US24047951500  6/9/2020      Maryland   Tract 9515, Worcester   
49251   1400000US24047951700  6/9/2020      Maryland   Tract 9517, Worcester   
49254   1400000US24510010300  6/9/2020      Maryland    Tract 103, Baltimore   
123248  1400000US72153750602  6/9/2020   Puerto Rico    Tract 7506.02, Yauco   

         Region      Geo_Type  DRRINT  DRRALL  CRRINT  not_int  CRRALL  
100700  Midwest  Census Tract     0.1    10.6    19.1     46.3    65.4  
26189     South  Census Tract     0.0     9.4    17.0     50.3    67.3  
24103     South  Census Tract     0.0     8.3    24.5     51.7    76.2  
42486   Midwest  Census Tract     0.0     7.9    26.2     42.3    68.5  
100720  Midwest  Census Tract     0.1     7.9    17.8     40.5    58.3  
...         ...           ...     ...     ...     ...      ...     ...  
49248     South  Census Tract     0.0     0.0    47.3     13.2    60.5  
49250     South  Census Tract     0.0     0.0    46.6     13.3    59.9  
49251     South  Census Tract     0.0     0.0    31.8      6.5    38.3  
49254     South  Census Tract     0.0     0.0    61.3      4.1    65.4  
123248    South  Census Tract     0.0     0.0     4.6      0.6     5.2  

[84519 rows x 11 columns]

In [115]:
# highest cumulative response rate (CRRALL)
tracts.sort_values(by='CRRALL', ascending=False)

GEO_ID RESP_DATE          State                Geo_Name  \
27200   1400000US13215010606  6/9/2020        Georgia  Tract 106.06, Muscogee   
50423   1400000US25013812903  6/9/2020  Massachusetts  Tract 8129.03, Hampden   
54026   1400000US26099223801  6/9/2020       Michigan   Tract 2238.01, Macomb   
54838   1400000US26139021605  6/9/2020       Michigan    Tract 216.05, Ottawa   
113809  1400000US51059492202  6/9/2020       Virginia  Tract 4922.02, Fairfax   
...                      ...       ...            ...                     ...   
54231   1400000US26099982100  6/9/2020       Michigan      Tract 9821, Macomb   
3046    1400000US04005980000  6/9/2020        Arizona    Tract 9800, Coconino   
483      2560000US1995T00100  6/9/2020            NaN       T001, Los Coyotes   
497      2560000US2190T00100  6/9/2020            NaN       T001, Mesa Grande   
4143    1400000US04017940010  6/9/2020        Arizona   Tract 9400.10, Navajo   

         Region      Geo_Type  DRRINT  DRRALL  CRRINT  not_int  CRRALL  
27200     South  Census Tract     0.0     0.0     0.0     98.1    98.1  
50423   Midwest  Census Tract     0.0     0.0    86.2      6.9    93.1  
54026   Midwest  Census Tract     0.0     0.0    86.1      6.5    92.6  
54838   Midwest  Census Tract     0.0     0.0    85.4      7.0    92.4  
113809    South  Census Tract     0.0     0.0    88.9      3.3    92.2  
...         ...           ...     ...     ...     ...      ...     ...  
54231   Midwest  Census Tract     0.0     0.0     0.0      0.0     0.0  
3046       West  Census Tract     0.0     0.0     0.0      0.0     0.0  
483          na  Tribal Tract     0.0     0.0     0.0      0.0     0.0  
497          na  Tribal Tract     0.0     0.0     0.0      0.0     0.0  
4143       West  Census Tract     0.0     0.0     0.0      0.0     0.0  

[84519 rows x 11 columns]

#### Tribal tracts

In [116]:
# response rates in tribal tracts
tribal = tracts[tracts['Geo_Type'].str.contains("Tribal")]
tribal.sort_values(by='CRRALL', ascending=False)

GEO_ID RESP_DATE State                 Geo_Name Region  \
566  2560000US2560T00400  6/9/2020   NaN        T004, Oneida (WI)     na   
452  2560000US1610T00600  6/9/2020   NaN           T006, Isabella     na   
565  2560000US2560T00300  6/9/2020   NaN        T003, Oneida (WI)     na   
604  2560000US3000T00200  6/9/2020   NaN           T002, Puyallup     na   
564  2560000US2560T00200  6/9/2020   NaN        T002, Oneida (WI)     na   
..                   ...       ...   ...                      ...    ...   
497  2560000US2190T00100  6/9/2020   NaN        T001, Mesa Grande     na   
494  2560000US2147T00100  6/9/2020   NaN  T001, Mashpee Wampanoag     na   
483  2560000US1995T00100  6/9/2020   NaN        T001, Los Coyotes     na   
582  2560000US2695T00100  6/9/2020   NaN      T001, Passamaquoddy     na   
433  2560000US1440T00100  6/9/2020   NaN          T001, Havasupai     na   

         Geo_Type  DRRINT  DRRALL  CRRINT  not_int  CRRALL  
566  Tribal Tract     0.0     0.1    78.7      6.7    85.4  
452  Tribal Tract     0.0     0.0    69.0     10.6    79.6  
565  Tribal Tract     0.0     0.0    68.4      9.7    78.1  
604  Tribal Tract     0.1     0.1    71.3      6.2    77.5  
564  Tribal Tract     0.1     0.1    67.1      9.5    76.6  
..            ...     ...     ...     ...      ...     ...  
497  Tribal Tract     0.0     0.0     0.0      0.0     0.0  
494  Tribal Tract     0.0     0.0     0.0      0.0     0.0  
483  Tribal Tract     0.0     0.0     0.0      0.0     0.0  
582  Tribal Tract     0.0     0.0     0.0      0.0     0.0  
433  Tribal Tract     0.0     0.0     0.0      0.0     0.0  

[426 rows x 11 columns]

In [64]:
### tribal areas and tracts stats

#mean non internet response
tribal.mean(axis=0)['not_int']

6.115962441314556

In [65]:
# mean internet response rate
tribal.mean(axis=0)['CRRINT']

19.630281690140855

In [66]:
# mean overall response rate
tribal.mean(axis=0)['CRRALL']

25.7462441314554

#### Tracts with 0 overall response rate

In [117]:
## Places with 0 cumulative response rate
is_zero = tracts['CRRALL'] == 0.0
zeros = tracts[is_zero]
zeros

GEO_ID RESP_DATE         State  \
433      2560000US1440T00100  6/9/2020           NaN   
483      2560000US1995T00100  6/9/2020           NaN   
494      2560000US2147T00100  6/9/2020           NaN   
497      2560000US2190T00100  6/9/2020           NaN   
529      2560000US2430T01800  6/9/2020           NaN   
582      2560000US2695T00100  6/9/2020           NaN   
2959    1400000US04001944202  6/9/2020       Arizona   
2961    1400000US04001944302  6/9/2020       Arizona   
3041    1400000US04005942202  6/9/2020       Arizona   
3046    1400000US04005980000  6/9/2020       Arizona   
4142    1400000US04017940008  6/9/2020       Arizona   
4143    1400000US04017940010  6/9/2020       Arizona   
8661    1400000US06037277400  6/9/2020    California   
10086   1400000US06037980020  6/9/2020    California   
14006   1400000US06083980100  6/9/2020    California   
18716   1400000US10001980000  6/9/2020      Delaware   
19336   1400000US12005000700  6/9/2020       Florida   
46742   1400000US22125951703  6/9/2020     Louisiana   
53866   1400000US26083980100  6/9/2020      Michigan   
54231   1400000US26099982100  6/9/2020      Michigan   
73868   1400000US36047005303  6/9/2020      New York   
75154   1400000US36061000500  6/9/2020      New York   
75965   1400000US36081000104  6/9/2020      New York   
89622   1400000US40109980003  6/9/2020      Oklahoma   
89623   1400000US40109980005  6/9/2020      Oklahoma   
94380   1400000US42019980100  6/9/2020  Pennsylvania   
101649  1400000US47037980100  6/9/2020     Tennessee   
122646  1400000US72049950500  6/9/2020   Puerto Rico   

                             Geo_Name     Region      Geo_Type  DRRINT  \
433                   T001, Havasupai         na  Tribal Tract     0.0   
483                 T001, Los Coyotes         na  Tribal Tract     0.0   
494           T001, Mashpee Wampanoag         na  Tribal Tract     0.0   
497                 T001, Mesa Grande         na  Tribal Tract     0.0   
529               T018, Navajo Nation         na  Tribal Tract     0.0   
582               T001, Passamaquoddy         na  Tribal Tract     0.0   
2959            Tract 9442.02, Apache       West  Census Tract     0.0   
2961            Tract 9443.02, Apache       West  Census Tract     0.0   
3041          Tract 9422.02, Coconino       West  Census Tract     0.0   
3046             Tract 9800, Coconino       West  Census Tract     0.0   
4142            Tract 9400.08, Navajo       West  Census Tract     0.0   
4143            Tract 9400.10, Navajo       West  Census Tract     0.0   
8661          Tract 2774, Los Angeles       West  Census Tract     0.0   
10086      Tract 9800.20, Los Angeles       West  Census Tract     0.0   
14006       Tract 9801, Santa Barbara       West  Census Tract     0.0   
18716                Tract 9800, Kent      South  Census Tract     0.0   
19336                    Tract 7, Bay      South  Census Tract     0.0   
46742   Tract 9517.03, West Feliciana      South  Census Tract     0.0   
53866            Tract 9801, Keweenaw    Midwest  Census Tract     0.0   
54231              Tract 9821, Macomb    Midwest  Census Tract     0.0   
73868              Tract 53.03, Kings  Northeast  Census Tract     0.0   
75154               Tract 5, New York  Northeast  Census Tract     0.0   
75965              Tract 1.04, Queens  Northeast  Census Tract     0.0   
89622         Tract 9800.03, Oklahoma      South  Census Tract     0.0   
89623         Tract 9800.05, Oklahoma      South  Census Tract     0.0   
94380              Tract 9801, Butler  Northeast  Census Tract     0.0   
101649           Tract 9801, Davidson      South  Census Tract     0.0   
122646            Tract 9505, Culebra      South  Census Tract     0.0   

        DRRALL  CRRINT  not_int  CRRALL  
433        0.0     0.0      0.0     0.0  
483        0.0     0.0      0.0     0.0  
494        0.0     0.0      0.0     0.0  
497        0.0     0.0      0.0     0.0  
529        0.0     0.0      0.0     0.0 

## States

In [137]:
# response rate by state
is_states = merged['Geo_Type'] == 'State'
states = merged[is_states]
states

# highest cumulative response rate (CRRALL)
states.sort_values(by='CRRALL', ascending=False)

GEO_ID RESP_DATE                 State              Geo_Name  \
56336   0400000US27  6/9/2020             Minnesota             Minnesota   
118515  0400000US55  6/9/2020             Wisconsin             Wisconsin   
38820   0400000US19  6/9/2020                  Iowa                  Iowa   
51572   0400000US26  6/9/2020              Michigan              Michigan   
65274   0400000US31  6/9/2020              Nebraska              Nebraska   
115591  0400000US53  6/9/2020            Washington            Washington   
113065  0400000US51  6/9/2020              Virginia              Virginia   
83303   0400000US39  6/9/2020                  Ohio                  Ohio   
29374   0400000US17  6/9/2020              Illinois              Illinois   
35465   0400000US18  6/9/2020               Indiana               Indiana   
47969   0400000US24  6/9/2020              Maryland              Maryland   
40744   0400000US20  6/9/2020                Kansas                Kansas   
17610   0400000US09  6/9/2020           Connecticut           Connecticut   
111588  0400000US49  6/9/2020                  Utah                  Utah   
43476   0400000US21  6/9/2020              Kentucky              Kentucky   
91922   0400000US42  6/9/2020          Pennsylvania          Pennsylvania   
90648   0400000US41  6/9/2020                Oregon                Oregon   
15838   0400000US08  6/9/2020              Colorado              Colorado   
68178   0400000US34  6/9/2020            New Jersey            New Jersey   
28672   0400000US16  6/9/2020                 Idaho                 Idaho   
49614   0400000US25  6/9/2020         Massachusetts         Massachusetts   
6189    0400000US06  6/9/2020            California            California   
61794   0400000US29  6/9/2020              Missouri              Missouri   
101175  0400000US47  6/9/2020             Tennessee             Tennessee   
99884   0400000US46  6/9/2020          South Dakota          South Dakota   
66773   0400000US32  6/9/2020                Nevada                Nevada   
81830   0400000US38  6/9/2020          North Dakota          North Dakota   
67583   0400000US33  6/9/2020         New Hampshire         New Hampshire   
18672   0400000US10  6/9/2020              Delaware              Delaware   
97959   0400000US44  6/9/2020          Rhode Island          Rhode Island   
746     0400000US01  6/9/2020               Alabama               Alabama   
19197   0400000US12  6/9/2020               Florida               Florida   
78534   0400000US37  6/9/2020        North Carolina        North Carolina   
18991   0400000US11  6/9/2020  District of Columbia  District of Columbia   
2938    0400000US04  6/9/2020               Arizona               Arizona   
24742   0400000US13  6/9/2020               Georgia               Georgia   
71666   0400000US36  6/9/2020              New York              New York   
60537   0400000US28  6/9/2020           Mississippi           Mississippi   
28242   0400000US15  6/9/2020                Hawaii                Hawaii   
98262   0400000US45  6/9/2020        South Carolina        South Carolina   
103304  0400000US48  6/9/2020                 Texas                 Texas   
4788    0400000US05  6/9/2020              Arkansas              Arkansas   
45317   0400000US22  6/9/2020             Louisiana             Louisiana   
88791   0400000US40  6/9/2020              Oklahoma              Oklahoma   
64770   0400000US30  6/9/2020               Montana               Montana   
112577  0400000US50  6/9/2020               Vermont               Vermont   
47063   0400000US23  6/9/2020                 Maine                 Maine   
121968  0400000US56  6/9/2020               Wyoming               Wyoming   
70937   0400000US35  6/9/2020            New Mexico            New Mexico   
117679  0400000US54  6/9/2020         West Virginia         West Virginia   
2711    0400000US02  6/9/2020                Alaska                Alaska   
122249  0

In [119]:
# highest non-internet response rate (not_int)
states.sort_values(by='not_int', ascending=False)

GEO_ID RESP_DATE                 State              Geo_Name  \
60537   0400000US28  6/9/2020           Mississippi           Mississippi   
4788    0400000US05  6/9/2020              Arkansas              Arkansas   
746     0400000US01  6/9/2020               Alabama               Alabama   
43476   0400000US21  6/9/2020              Kentucky              Kentucky   
101175  0400000US47  6/9/2020             Tennessee             Tennessee   
45317   0400000US22  6/9/2020             Louisiana             Louisiana   
35465   0400000US18  6/9/2020               Indiana               Indiana   
83303   0400000US39  6/9/2020                  Ohio                  Ohio   
98262   0400000US45  6/9/2020        South Carolina        South Carolina   
51572   0400000US26  6/9/2020              Michigan              Michigan   
117679  0400000US54  6/9/2020         West Virginia         West Virginia   
38820   0400000US19  6/9/2020                  Iowa                  Iowa   
61794   0400000US29  6/9/2020              Missouri              Missouri   
88791   0400000US40  6/9/2020              Oklahoma              Oklahoma   
40744   0400000US20  6/9/2020                Kansas                Kansas   
65274   0400000US31  6/9/2020              Nebraska              Nebraska   
91922   0400000US42  6/9/2020          Pennsylvania          Pennsylvania   
47063   0400000US23  6/9/2020                 Maine                 Maine   
19197   0400000US12  6/9/2020               Florida               Florida   
78534   0400000US37  6/9/2020        North Carolina        North Carolina   
97959   0400000US44  6/9/2020          Rhode Island          Rhode Island   
29374   0400000US17  6/9/2020              Illinois              Illinois   
99884   0400000US46  6/9/2020          South Dakota          South Dakota   
118515  0400000US55  6/9/2020             Wisconsin             Wisconsin   
113065  0400000US51  6/9/2020              Virginia              Virginia   
71666   0400000US36  6/9/2020              New York              New York   
24742   0400000US13  6/9/2020               Georgia               Georgia   
17610   0400000US09  6/9/2020           Connecticut           Connecticut   
64770   0400000US30  6/9/2020               Montana               Montana   
56336   0400000US27  6/9/2020             Minnesota             Minnesota   
81830   0400000US38  6/9/2020          North Dakota          North Dakota   
18672   0400000US10  6/9/2020              Delaware              Delaware   
112577  0400000US50  6/9/2020               Vermont               Vermont   
90648   0400000US41  6/9/2020                Oregon                Oregon   
70937   0400000US35  6/9/2020            New Mexico            New Mexico   
6189    0400000US06  6/9/2020            California            California   
28242   0400000US15  6/9/2020                Hawaii                Hawaii   
103304  0400000US48  6/9/2020                 Texas                 Texas   
2938    0400000US04  6/9/2020               Arizona               Arizona   
66773   0400000US32  6/9/2020                Nevada                Nevada   
49614   0400000US25  6/9/2020         Massachusetts         Massachusetts   
68178   0400000US34  6/9/2020            New Jersey            New Jersey   
47969   0400000US24  6/9/2020              Maryland              Maryland   
28672   0400000US16  6/9/2020                 Idaho                 Idaho   
67583   0400000US33  6/9/2020         New Hampshire         New Hampshire   
121968  0400000US56  6/9/2020               Wyoming               Wyoming   
115591  0400000US53  6/9/2020            Washington            Washington   
15838   0400000US08  6/9/2020              Colorado              Colorado   
18991   0400000US11  6/9/2020  District of Columbia  District of Columbia   
2711    0400000US02  6/9/2020                Alaska                Alaska   
111588  0400000US49  6/9/2020                  Utah                  Utah   
122249  0

#### State stats

In [48]:
states.mean(axis=0)['not_int']

11.63076923076923

In [49]:
states.mean(axis=0)['CRRINT']

47.101923076923065

In [50]:
states.mean(axis=0)['CRRALL']

58.73269230769233

In [141]:
# average by region
# NOTE as tribal tracts are not assigned to a state they do not have a corresponding region and thus are not counted in the regional calculations
states.groupby('Region').mean().sort_values(by='CRRALL', ascending=False)

DRRINT    DRRALL     CRRINT    not_int     CRRALL
Region                                                      
Midwest       0.0  0.330769  52.238462  12.815385  65.053846
Northeast     0.0  0.100000  47.337500  11.200000  58.537500
West          0.0  0.169231  48.646154   8.846154  57.492308
South         0.0  0.266667  42.172222  12.977778  55.150000

#### State stats without Puerto Rico

In [120]:
no_pr = states[states.State != 'Puerto Rico']

# average non internet response rate
no_pr.mean(axis=0)['not_int']

11.84705882352941

In [ ]:
# average internet response
no_pr.mean(axis=0)['CRRINT']

In [ ]:
# average overall response rate
no_pr.mean(axis=0)['CRRALL']

In [142]:
# average region response rate (so south now excludes Puerto Rico)
no_pr.groupby('Region').mean().sort_values(by='CRRALL', ascending=False)

DRRINT    DRRALL     CRRINT    not_int     CRRALL
Region                                                      
Midwest       0.0  0.330769  52.238462  12.815385  65.053846
Northeast     0.0  0.100000  47.337500  11.200000  58.537500
South         0.0  0.282353  44.152941  13.705882  57.858824
West          0.0  0.169231  48.646154   8.846154  57.492308

## Compare to 2010 response rates

### 2010 to 2020 State Response Differences

In [163]:
# read in csvs with 2010 response data for tracts and states
rates2010 = pd.read_csv('2010responserate.csv')
rates2010

NAME            county         State  \
0          Census Tract 201    Autauga County       Alabama   
1          Census Tract 202    Autauga County       Alabama   
2          Census Tract 203    Autauga County       Alabama   
3          Census Tract 204    Autauga County       Alabama   
4       Census Tract 205.01    Autauga County       Alabama   
...                     ...               ...           ...   
84514  Census Tract 7505.01   Yauco Municipio   Puerto Rico   
84515  Census Tract 7505.02   Yauco Municipio   Puerto Rico   
84516  Census Tract 7505.03   Yauco Municipio   Puerto Rico   
84517  Census Tract 7506.01   Yauco Municipio   Puerto Rico   
84518  Census Tract 7506.02   Yauco Municipio   Puerto Rico   

                     GEO_ID  FSRR2010  state_num  county_num   tract  
0      1400000US01001020100      70.6          1           1   20100  
1      1400000US01001020200      70.1          1           1   20200  
2      1400000US01001020300      73.6          1           1   20300  
3      1400000US01001020400      78.4          1           1   20400  
4      1400000US01001020501      81.2          1           1   20501  
...                     ...       ...        ...         ...     ...  
84514  1400000US72153750501      70.4         72         153  750501  
84515  1400000US72153750502      73.4         72         153  750502  
84516  1400000US72153750503      63.4         72         153  750503  
84517  1400000US72153750601      67.1         72         153  750601  
84518  1400000US72153750602      57.8         72         153  750602  

[84519 rows x 8 columns]

In [161]:
## difference in row numbers: both tract dfs have 84519 rows, but when joined only 84093
# Identify what values are in rates2010 and not in tracts
key_diff1 = set(rates2010.GEO_ID).difference(tracts.GEO_ID)
len(key_diff1)
key_diff1

# Identify what values are in tracts and not in rates2010
key_diff2 = set(tracts.GEO_ID).difference(rates2010.GEO_ID)
len(key_diff2)
key_diff2

426

##### Conclusion: 2010 rates do not include tribal tracts while 2020 tracts are missing some tracts in a multitude of states, likely due to a change in tract boundaries. Those differences account for 426 tracts, which is .5% of the original 84519 tracts

In [164]:
# merge with 2020 tracts
tracts_merged = pd.merge(tracts, rates2010, on='GEO_ID')
cols = tracts_merged.columns.tolist()
cols = ['GEO_ID','NAME','county', 'State_y', 'Region', 'CRRALL', 'FSRR2010']
tracts_merged = tracts_merged[cols]
tracts_merged['10-20 difference'] = tracts_merged['FSRR2010'] - tracts_merged['CRRALL']
tracts_merged.sort_values(by='10-20 difference', ascending=False)

GEO_ID                  NAME                  county  \
8287   1400000US06053980000     Census Tract 9800         Monterey County   
53795  1400000US36103159510  Census Tract 1595.10          Suffolk County   
7980   1400000US06037980021  Census Tract 9800.21      Los Angeles County   
49059  1400000US36001000404     Census Tract 4.04           Albany County   
10379  1400000US06071010906   Census Tract 109.06   San Bernardino County   
...                     ...                   ...                     ...   
83055  1400000US56013940202  Census Tract 9402.02          Fremont County   
83056  1400000US56013940301  Census Tract 9403.01          Fremont County   
83057  1400000US56013940302  Census Tract 9403.02          Fremont County   
83058  1400000US56013940400     Census Tract 9404          Fremont County   
83059  1400000US56013940500     Census Tract 9405          Fremont County   

           State_y     Region  CRRALL  FSRR2010  10-20 difference  
8287    California       West     0.7     100.0              99.3  
53795     New York  Northeast     0.8     100.0              99.2  
7980    California       West     1.6     100.0              98.4  
49059     New York  Northeast     1.8     100.0              98.2  
10379   California       West     2.9     100.0              97.1  
...            ...        ...     ...       ...               ...  
83055      Wyoming       West    17.9       NaN               NaN  
83056      Wyoming       West    21.5       NaN               NaN  
83057      Wyoming       West    31.6       NaN               NaN  
83058      Wyoming       West    43.1       NaN               NaN  
83059      Wyoming       West    30.4       NaN               NaN  

[84093 rows x 8 columns]

In [165]:
# average difference as of 6/9/20
tracts_merged.mean(axis=0)['10-20 difference']

7.510194825399082

In [ ]:
# average difference by region
states_merged.groupby('Region').mean()

### 2010 to 2020 State Response Differences

In [131]:
# read in csvs with 2010 response data for states
states2010 = pd.read_csv('states2010.csv')

# merge with 2020 states
states_merged = pd.merge(states, states2010, on='State')
cols = states_merged.columns.tolist()
cols = ['GEO_ID', 'State', 'Region',
 'CRRALL', 'response2000', 'response2010']
states_merged = states_merged[cols]
states_merged['10-20 difference'] = states_merged['response2010'] - states_merged['CRRALL']
states_merged.sort_values(by='10-20 difference', ascending=False)

GEO_ID                 State     Region  CRRALL  response2000  \
51  0400000US72           Puerto Rico      South     9.1          54.0   
1   0400000US02                Alaska       West    41.5          67.0   
40  0400000US45        South Carolina      South    55.6          68.0   
33  0400000US37        North Carolina      South    57.1          69.0   
50  0400000US56               Wyoming       West    50.8          75.0   
45  0400000US50               Vermont  Northeast    51.5          68.0   
19  0400000US23                 Maine  Northeast    50.9          67.0   
48  0400000US54         West Virginia      South    48.2          68.0   
31  0400000US35            New Mexico       West    48.4          68.0   
26  0400000US30               Montana       West    51.8          72.0   
41  0400000US46          South Dakota    Midwest    60.3          80.0   
9   0400000US12               Florida      South    58.3          71.0   
43  0400000US48                 Texas      South    55.5          70.0   
42  0400000US47             Tennessee      South    60.8          72.0   
10  0400000US13               Georgia      South    57.0          72.0   
8   0400000US11  District of Columbia      South    57.1          69.0   
34  0400000US38          North Dakota    Midwest    59.7          78.0   
39  0400000US44          Rhode Island  Northeast    58.9          73.0   
49  0400000US55             Wisconsin    Midwest    67.9          85.0   
36  0400000US40              Oklahoma      South    54.3          71.0   
3   0400000US05              Arkansas      South    55.4          70.0   
14  0400000US18               Indiana    Midwest    65.5          79.0   
29  0400000US33         New Hampshire  Northeast    59.6          74.0   
38  0400000US42          Pennsylvania  Northeast    63.8          78.0   
0   0400000US01               Alabama      South    58.8          68.0   
25  0400000US29              Missouri    Midwest    61.0          76.0   
12  0400000US16                 Idaho       West    63.0          77.0   
24  0400000US28           Mississippi      South    56.0          70.0   
32  0400000US36              New York  Northeast    56.1          69.0   
7   0400000US10              Delaware      South    59.1          71.0   
17  0400000US21              Kentucky      South    64.3          72.0   
37  0400000US41                Oregon       West    63.5          77.0   
11  0400000US15                Hawaii       West    55.6          66.0   
35  0400000US39                  Ohio    Midwest    65.8          79.0   
21  0400000US25         Massachusetts    Midwest    62.9          76.0   
46  0400000US51              Virginia      South    65.9          76.0   
2   0400000US04               Arizona       West    57.1          70.0   
15  0400000US19                  Iowa    Midwest    67.4          82.0   
6   0400000US09           Connecticut  Northeast    64.5          78.0   
16  0400000US20                Kansas    Midwest    64.7          78.0   
4   0400000US06            California       West    62.0          76.0   
30  0400000US34            New Jersey  Northeast    63.0          76.0   
28  0400000US32                Nevada       West    60.0          71.0   
20  0400000US24              Maryland      South    65.2          76.0   
44  0400000US49                  Utah       West    64.3          74.0   
22  0400000US26              Michigan    Midwest    67.3          79.0   
23  0400000US27             Minnesota    Midwest    70.5          81.0   
13  0400000US17              Illinois    Midwest    65.8          76.0   
27  0400000US31              Nebraska    Midwest    66.9          81.0   
18  0400000US22             Louisiana      South    55.0          68.0   
47  0400000US53            Washington       West    66.1          75.0   
5   0400000US08              Colorado       West    63.3          75.0   

    response2010  10-20 difference  
51          54.0              44.9  
1           

In [130]:
# average difference as of 6/9/20
states_merged.mean(axis=0)['10-20 difference']

14.094230769230771

In [133]:
# average difference by region
states_merged.groupby('Region').mean()

CRRALL  response2000  response2010  10-20 difference
Region                                                            
Midwest    65.053846     79.230769     77.307692         12.253846
Northeast  58.537500     72.875000     72.375000         13.837500
South      55.150000     69.722222     71.166667         16.016667
West       57.492308     72.538462     70.923077         13.430769